### ISSUE TITLE
Additional support for third party instance funding

### ISSUE THREAD

**Issue by wojake** [EvernodeXRPL/evernode-host#89]
Right now, tenants are the only party that are allowed to fund their instances individually. Considering how complex things may get with nomad dApps and their internal structures & possible bugs. It would be best to consider for us allowing third party entities to fund a tenant's instance.

2 funding options: `open` and `closed`
### open
Anyone may pay the host to fund the tenant's instance.

### closed
No one except for the tenant could pay the host for its instance.

The tenant could decide which option it'll pick during initial instance acquisition.

**Comment by chalith** [comment]
@wojake that's a good suggestion to enable in the future

**Comment by wojake** [comment]
https://twitter.com/whalewire/status/1666495467520155648?s=46 Another reason to have this feature live during launch

**Comment by ravinsp** [comment]
Just noting down here (for future reference), that there could be a third option, where current funder, tenant or the contract can nominate a different funder.

**Comment by wojake** [comment]
I like that, +1

**Comment by marklikeapenny** [comment]
"GoFundMe" Hook ?
Not just but also for Evernode tenants.

**Comment by marklikeapenny** [comment]
Does this already cover handling of not used up funds?
Tenant shuts down instance before all foreign funds used up. Runs away with excess funds.

**Comment by ravinsp** [comment]
I don't think we will go down the path of refunds for life extensions. If you send your funds to extend instance lifetime that's your decision. Choosing to prematurely delete the instance is also your decision and should not be tied with any refund mechanism. Only time we would support refund is during instance creation. If the instance creation fails, host would refund.

**Comment by marklikeapenny** [comment]
Sorry I used the wrong party in my comment above.

This scenario:
- Tenant pre-pays some larger amount because he intends his smart contract to run for a long time.
- Malicious hoster receives the large payment. Shuts down host. Runs away with funds.

Instead: Pay-per-use; Streaming payments with payment channels ?

**Comment by ravinsp** [comment]
@marklikeapenny we have already covered this argument some time back. It you don't trust the host, it's up to you to keep extending the lease by small durations (the smallest unit is 1 Moment).
